core/controllers/acl_decorators.py

```diff
@@ -521,9 +522,6 @@ def test_can_upload(self, **kwargs):
             UnauthorizedUserException: The user does not have
                 credentials to upload an exploration.
         """
-        if not self.user_id:
```

Comment:
Why delete?

Reply:
The next line(`current_user_services.is_current_user_super_admin()`) checks for the case if a user is guest or not also, so I removed this line.

Reply:
But you shouldn't, because they raise different exceptions.

Reply:
Done